Jscbc 813 user management snippets

osfameron opened the pull request:
draft PR pending https://issues.couchbase.com/browse/JSCBC-944

osfameron committed:
JSCBC-813 DOC-5314 user management snippet 1

osfameron committed:
JSCBC-813 user management snippet 2

osfameron committed:
JSCBC-813 remaining user-management examples

osfameron committed:
JSCBC-813 enable in nav

maria-robobug commented on modules/howtos/examples/user-manager.js:
```diff
@@ -0,0 +1,98 @@
+
+go()
+    .then((res) => console.log("DONE:", res))
+    .catch((err) => console.error("ERR:", err))
```
I think this will only `log` any error that might occur when running the example.  So it won't fail with an exit status, which is what I've been checking to assert if the code sample fails or passes in our automation.

Think we can just remove the `catch` line here, but let me know if you have other thoughts.

osfameron replied:
Yeah, that's formatted to the template we've previously used :)

Happy for it to be changed subsequently in the automation work to whatever makes sense?
(I can change here if you prefer, but thought might be cleaner to do where that kind of tweak is being done anyway?)

osfameron replied:
Ah, looks like your automation stuff may go in sooner than this, in which case I can tidy it up to that before committing (good practice for me!)

osfameron commented:
I've updated couchnode as per @brett19 and confirmed that JSCBC-944 is fixed, yay.

osfameron committed:
JSCBC-813 add test for user-manager.js snippet

maria-robobug commented on modules/howtos/examples/user-manager.js:
```diff
@@ -0,0 +1,97 @@
+const couchbase = require("couchbase");
+
+async function go() {
+    const clusterAdm = new couchbase.Cluster("couchbase://localhost", {
```
Probably best to use the new `couchbase.connect` method instead fo the deprecated `couchbase.Cluster` no?

So `await couchbase.connect("couchbase://localhost")`

maria-robobug commented on modules/howtos/examples/user-manager.js:
```diff
@@ -0,0 +1,97 @@
+    // the username and password already assigned by the administrator
+
+    // tag::usermanagement_3[]
+    const userCluster = new couchbase.Cluster(
```
And here also `couchbase.connect`

maria-robobug commented on modules/test/test-howtos.bats:
```diff
@@ -116,6 +116,11 @@ mutate-increment
 EOF
 }
 
+@test "[howtos] - user-manager.js" {
```
🥳 wahoo!

maria-robobug reviewed (approved):
👍 , just minor comment on updating to the new `couchbase.connect` function.

osfameron committed:
JSCBC-813 modern `await` syntax

Also ensure Node terminates correctly by catching and logging
any error, and then returning an error.
